flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/TableStore.java

```diff
@@ -175,28 +181,33 @@ private MergeEngine mergeEngine() {
         return options.get(MERGE_ENGINE);
     }
 
-    private FileStore buildFileStore() {
+    FileStore buildFileStore() {
         RowType partitionType = TypeUtils.project(type, partitionKeysIndex());
         FileStoreOptions fileStoreOptions = new FileStoreOptions(options);
+        String json = options.get(COMPACTION_SCANNED_MANIFEST);
+        PartitionedManifestMeta manifestMeta =
+                json == null ? null : PartitionedManifestMeta.fromJson(json);
         int[] trimmedPrimaryKeys = trimmedPrimaryKeysIndex();
         if (trimmedPrimaryKeys.length == 0) {
             return FileStoreImpl.createWithValueCount(
-                    fileStoreOptions.path(tableIdentifier).toString(),
-                    schema.id(),
-                    fileStoreOptions,
-                    user,
-                    partitionType,
-                    type);
+                            fileStoreOptions.path(tableIdentifier).toString(),
+                            schema.id(),
+                            fileStoreOptions,
+                            user,
+                            partitionType,
+                            type)
+                    .withPartitionedMeta(manifestMeta);
```

Comment:
Nit: could we just pass the `manifestMeta` as a constructor argument ?   All arguments are constructor arguments except this `manifestMeta`,  looks strange for me.

Reply:
> Nit: could we just pass the manifestMeta as a constructor argument ? All arguments are constructor arguments except this manifestMeta, looks strange for me.

I'm okay with both. I just thought the param list was too long, and this field is not required for every condition.